samples/19.customization-plain-ui/README.md

```diff
@@ -0,0 +1,53 @@
+# Customizing Web Chat with a plain HTML UI
+
+This sample shows how to use Web Chat without its canned UI component.
+
+When we design Web Chat, we think deeply about customizations. To make customizations great, we design our component in layers. Developers can go into the full experience of Web Chat by using all of our layers as-is. Or they could build their own UI from ground up with our business logic. Or they can opt into our UIs but replacing just a handful of components.
+
+Below is the explanation of different layers of Web Chat.
+
+- [botframework-directlinejs](https://github.com/Microsoft/BotFramework-DirectLineJS) should be eventful and stateless
+- [botframework-webchat-core](https://github.com/Microsoft/BotFramework-WebChat/tree/master/packages/core) (a.k.a. Web Chat Core) is using Redux, provide stateful data store but minimal to support various UI frameworks and app scenarios
+   - UI framework agnostic: we think about building apps on React Native and Angular (using Redux bindings)
+- [botframework-webchat-component](https://github.com/Microsoft/BotFramework-WebChat/tree/master/packages/component) (a.k.a. Web Chat UI) has multiple layers
+   - [`Composer`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/component/src/Composer.js) is the React context which connect to Web Chat Core. If you want to build a component using Web Chat, you will need to put the component your inside `<Composer>`. This is very similar to Redux `<Provider>` but using React Context instead
+   - [`connectToWebChat`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/component/src/connectToWebChat.js) should provide mechanisms to build components from ground up to support various creative ideas. `connectToWebChat` is very similar to Redux `connect` function, provides access to all data required to build a chat component
+   - [`MicrophoneButton.connectMicrophoneButton`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/component/src/SendBox/MicrophoneButton.js) should provide HOC function to hoist features from Web Chat as props to the connected component. Although this is very similar to `connectToWebChat`, it provide encapsulation of information and business logics required for the specific component
+   - [`MicrophoneButton`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/component/src/SendBox/MicrophoneButton.js) is the actual component developer can use to reproduce the full Web Chat experience
```

Comment:
Lines 9-16 might work better as a table or something.

Reply:
Converted to table format.